src/main/resources/userEmailTemplate-feedbackSessionClosing.html

```diff
@@ -66,8 +66,16 @@
 </p>
 
 <p>
-  <strong>No action is required if you have already submitted.</strong>
-  <br>If you encounter any problems when using the system, you can email TEAMMATES support team at ${supportEmail}.
+  <ul>
+    <li><strong>If already submitted:</strong></li>
+    No action is required.
+    <li><strong>Requests for deadline extensions,</strong></li>
+    <strong>problems regarding team/student data</strong> (e.g. wrong team, misspelled name),
+    <br>and <strong>other non-technical queries</strong> about the feedback session :
+    <br>Contact the instructors of the course: ${coOwnersEmails}
+    <li><strong>Technical help regarding TEAMMATES</strong>  (e.g. submission link is not working) :</li>
+    Email TEAMMATES support team at ${supportEmail}.
```

Comment:
I know this might be a trivial question, but how would I get the email? should I deploy the project in live server and create a session to get the email? :sweat_smile:

Reply:
That works, but for a much simpler alternative you can simply open this HTML file in a browser and see how it looks.

Reply:
Okay.

Reply:
**sessionClosingEmailForStudent:**<br>
![screenshot 251](https://user-images.githubusercontent.com/10681755/33388800-eaf0c382-d556-11e7-9fd2-e069d23bdf7f.png)

**sessionClosedEmailForStudent:**<br>
![screenshot 252](https://user-images.githubusercontent.com/10681755/33388836-046853b6-d557-11e7-8216-3e56dbe2a3ea.png)

Reply:
**Query:**
Should I remove this line from `userEmailTemplate-feedbackSessionClosing.html` (as the instructor's emails are already provided in the bullet points.): <br>
![screenshot 251 - copy](https://user-images.githubusercontent.com/10681755/33390271-6afea16c-d55b-11e7-890a-fee20eafab69.png)

One other thing that I wanted to ask is that, when I modified `userEmailFragment-sessionClosedAdditionalInstructionsFragment.html` and ran GodMode, four sessionClosed files were modified. But these two lines were not removed. <br>
![screenshot 252](https://user-images.githubusercontent.com/10681755/33389923-5cfd2116-d55a-11e7-976f-5073c6953429.png)
<br>
And when I tried to remove them (I know this is a big No No in GodMode document, but I created a new branch `tinker` and modified the test files), I ran into AssertionError `<expected>`  `<actual>`  error, (The diff was the last line) <br>

>If you need any help or clarification regarding TEAMMATES, please email us at ${support.email}

This line was not in the `<expected>` file but was in the `<actual>` file. (But how is this possible if I changed every test file to match the content in the email, I mean from where those two lines are generating from if it is not in any of the modified files?). Upon running GodMode again, those lines were re-added in the `sessionClosed` files and the tests ran successfully.

Reply:
> Should I remove this line from `userEmailTemplate-feedbackSessionClosing.html` (as the instructor's emails are already provided in the bullet points.):

I'm for it, @damithc your opinions? Also, there seems to be an excessive amount of line breaks in the text at least for me.

> But how is this possible if I changed every test file to match the content in the email, I mean from where those two lines are generating from if it is not in any of the modified files?

@sukanta-27 take a look at the lines you modified. Do you see "For any non-technical queries ... you can contact your instructor(s) at: ..." and "If you need ..., please email us at ..." as part of the lines you deleted?

Reply:
> I'm for it, @damithc your opinions?

Yup, can remove those redundant parts.

> Also, there seems to be an excessive amount of line breaks in the text at least for me.

OK to reduce unnecessary line breaks in the bullet points too.

Reply:
>Yup, can remove those redundant parts.

Okay, I am working on it. @damithc @wkurniawan07 

>take a look at the lines you modified. Do you see "For any non-technical queries ... you can contact your instructor(s) at: ..." and "If you need ..., please email us at ..." as part of the lines you deleted?

I deleted the line "If you need ..., please email us at ..." in the new branch in my machine (Just to experiment :sweat_smile: ) in these files : 

```
sessionClosedEmailForInstructor.html
sessionClosedEmailForStudent.html
sessionClosedEmailTestingSanitizationForInstructor.html
sessionClosedEmailTestingSanitizationForStudent.html
```
and the lines were not in the `userEmailFragment-sessionClosedAdditionalInstructionsFragment.html` file to begin with, still when I ran `EmailGeneratorTest` I got this.
![screenshot 254](https://user-images.githubusercontent.com/10681755/33433301-0e7380ce-d601-11e7-89e2-cf16e04e4131.png)

Anyway, I will figure it out. :smile:

Reply:
@wkurniawan07 I am sorry, I just realized it was a stupid question. It hadn't paid attention to the fact that `userEmailFragment-sessionClosedAdditionalInstructionsFragment.html` is the additional information for the email. The main Template is in the `userEmailTemplate-feedbackSession.html`. The two lines I mentioned were coming from that file. :sweat_smile: 

But now the problem is that when I am rephasing the file `userEmailTemplate-feedbackSession.html` to remove the redundant lines I mentioned all these files are getting modified : 
```
        modified:   src/test/resources/emails/sessionClosedEmailForInstructor.html
        modified:   src/test/resources/emails/sessionClosedEmailForStudent.html
        modified:   src/test/resources/emails/sessionClosedEmailTestingSanitizationForInstructor.html
        modified:   src/test/resources/emails/sessionClosedEmailTestingSanitizationForStudent.html
        modified:   src/test/resources/emails/sessionOpeningEmailForInstructor.html
        modified:   src/test/resources/emails/sessionOpeningEmailForStudent.html
        modified:   src/test/resources/emails/sessionOpeningEmailTestingSanitizationForInstructor.html
        modified:   src/test/resources/emails/sessionOpeningEmailTestingSanitzationForStudent.html
        modified:   src/test/resources/emails/sessionReminderEmailForInstructor.html
        modified:   src/test/resources/emails/sessionReminderEmailForStudent.html

```

For the sessionClosed test files (like `sessionClosedEmailForStudent.html` ) it is providing the **desired** output : 
![screenshot 256](https://user-images.githubusercontent.com/10681755/33437578-6645037e-d60e-11e7-8788-c7560e23b456.png)

But for other emails like `sessionOpeningEmailForInstructor.html` the contact information is not present (As these files are not using any other files for additional information) : 
![screenshot 257](https://user-images.githubusercontent.com/10681755/33437920-6a0a65ca-d60f-11e7-9567-6b9c2cf4b8dc.png)

I am looking for a solution. But as this may take some refactoring, any suggestions will be helpful  :sweat_smile:

Reply:
@wkurniawan07 **Found a solution:**
Changed the current code of [`userEmailTemplate-feedbackSession.html`](https://github.com/TEAMMATES/teammates/blob/master/src/main/resources/userEmailTemplate-feedbackSession.html)
to (below is the `git diff`)
```html
 </p>

 <p>
-  For any non-technical queries (e.g. assigned to the wrong team, misspelled name, request for deadline extension, etc.), you can contact your instructor(s) at: ${coOwnersEmails}.
-</p>
-
-${additionalInstructions}
-<p>
-  If you need any help or clarifications regarding TEAMMATES, please email us at ${supportEmail}.
+  <ul>
+    ${additionalInstructions}
+    <li><strong>Requests for deadline extensions, problems regarding team/student data</strong> (e.g. wrong team, misspelled name),
+    and <strong>other non-technical queries</strong> about the feedback session :
+    <br>Contact the instructors of the course: ${coOwnersEmails}</li>
+    <li><strong>Technical help regarding TEAMMATES</strong> (e.g. submission link is not working) :
+    Email TEAMMATES support team at ${supportEmail}.</li>
+  </ul>
 </p>

```
and `userEmailFragment-sessionClosedAdditionalInstructionsFragment.html` to : 

```html
-<p>
-  <ul>
-    <li><strong>If already submitted:</strong> No action is required.</li>
-    <li><strong>Requests for deadline extensions, problems regarding team/student data</strong> (e.g. wrong team, misspelled name),
-    and <strong>other non-technical queries</strong> about the feedback session :
-    <br>Contact the instructors of the course: ${coOwnersEmails}</li>
-    <li><strong>Technical help regarding TEAMMATES</strong> (e.g. submission link is not working) :
-    Email TEAMMATES support team at ${supportEmail}.</li>
-  </ul>
-</p>
+<li><strong>If already submitted:</strong> No action is required.</li>

```

Now the SessionClosed files look like this : 
![screenshot 256](https://user-images.githubusercontent.com/10681755/33440283-98cf3c2c-d615-11e7-9253-5feef8b0ad75.png)

and the other files: 
```
       sessionOpeningEmailForInstructor.html
       sessionOpeningEmailForStudent.html
       sessionOpeningEmailTestingSanitizationForInstructor.html
       sessionOpeningEmailTestingSanitzationForStudent.html
       sessionReminderEmailForInstructor.html
       sessionReminderEmailForStudent.html
```
has been changed to : 
![screenshot 258](https://user-images.githubusercontent.com/10681755/33440517-2cd1fdb0-d616-11e7-8916-3426ce9d49a1.png)

I think this can work as the above mentioned files has also been rephased to have the bullet points. :smiley: